### PR TITLE
feat(angular/sidebar): make sidebar collapsible

### LINKED ISSUE #2263: [Feature request]: Collapsible Sidebar
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/sbb-design-systems/sbb-angular/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/sbb-design-systems/sbb-angular/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/orgs/sbb-design-systems/projects/1) for an issue that matches the one I want to file, without success.

### Request type

Request for a new component

### Functionality

the sidebar (overlayed variant) should have a closeable action (top right). 

### Link to design proposal file

https://www.figma.com/file/97aIxaMIDQIEK5Pq4lkUlQ/Lean?type=design&node-id=503%3A76216&mode=design&t=ns7Oc2hBVLi03ggi-1

### PR DESCRIPTION
**Pull request by mhaertwig**
This allows to make the `SbbSidebar` collapsible. If collapsible, the sidebar is displayed in `over` mode (above the content) and can be closed by clicking the close button. The `toggleSidebar()` method can be used to open or collapse the sidebar.

Closes #2263

### CODE REVIEW

File: src/components-examples/angular/sidebar/collapsible-sidebar/collapsible-sidebar-example.html [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<div class="sidebar-frame">
+  <sbb-icon-sidebar-container>
+    <sbb-icon-sidebar [position]="position.value">
+      <a sbbIconSidebarItem label="Station" href="">
+        <sbb-icon svgIcon="station-small"></sbb-icon>
+      </a>
+      <a sbbIconSidebarItem label="Lockers" class="sbb-active" (click)="sidebar.toggle()">
```
Comment (mhaertwig) [inline src/components-examples/angular/sidebar/collapsible-sidebar/collapsible-sidebar-example.html:7]: @mcilurzo, @kyubisation, @jeripeierSBB this needs to be discussed.

The items of the `SbbIconSidebar` are links (`<a />` tags). The links are usually used to navigate to new pages having it's own `SbbSidebar`. In the demo I am (mis)using the active link to toggle the sidebar collapsed state.

I think that's not an optimal solution from UX and accessibility perspective. What do you think?

File: src/components-examples/angular/sidebar/collapsible-sidebar/collapsible-sidebar-example.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import { SbbIconModule } from '@sbb-esta/angular/icon';
+import { SbbInput } from '@sbb-esta/angular/input';
+import { SbbSelect } from '@sbb-esta/angular/select';
+import { SbbSidebarContainer, SbbSidebarModule } from '@sbb-esta/angular/sidebar';
+import { Subject } from 'rxjs';
+import { startWith, takeUntil } from 'rxjs/operators';
+
+/**
+ * @title Collapsible Sidebar
+ * @order 1
```
Comment (mhaertwig) [inline src/components-examples/angular/sidebar/collapsible-sidebar/collapsible-sidebar-example.ts:19]: - [ ] To be changed to 40

Comment (mhaertwig) [inline src/components-examples/angular/sidebar/collapsible-sidebar/collapsible-sidebar-example.ts:19]:
```suggestion
 * @order 40
```

File: src/components-examples/angular/sidebar/collapsible-sidebar/collapsible-sidebar-example.html (hunk 1 above)
Comment (kyubisation) [reply src/components-examples/angular/sidebar/collapsible-sidebar/collapsible-sidebar-example.html:7]: As discussed, I agree, but do not have a better solution.
An alternative might be to have a burger button in the content, but I'm not sure whether that works well.
@mcilurzo ?

**Review by kyubisation — APPROVED** [review]
LGTM 👍

**Pull request merged**